internal/runtimehooks/webhooks/server.go

```diff
@@ -78,68 +78,87 @@ func (s *Server) Start(ctx context.Context) error {
 		return err
 	}
 
-	// Create the ExtensionHandlers for the lifecycle hooks
-	lifecycleExtensionHandlers := lifecycle.NewExtensionHandlers(client)
-
-	// Register extension handlers.
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.BeforeClusterCreate,
-		Name:        "before-cluster-create",
-		HandlerFunc: lifecycleExtensionHandlers.DoBeforeClusterCreate,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
-	}
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.AfterControlPlaneInitialized,
-		Name:        "after-control-plane-initialized",
-		HandlerFunc: lifecycleExtensionHandlers.DoAfterControlPlaneInitialized,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
-	}
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.BeforeClusterUpgrade,
-		Name:        "before-cluster-upgrade",
-		HandlerFunc: lifecycleExtensionHandlers.DoBeforeClusterUpgrade,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
-	}
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.BeforeClusterDelete,
-		Name:        "before-cluster-delete",
-		HandlerFunc: lifecycleExtensionHandlers.DoBeforeClusterDelete,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
-	}
-
-	// Create the ExtensionHandlers for the topology mutation hooks
-	topologyMutationHandlers := mutation.NewExtensionHandlers(client)
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.DiscoverVariables,
-		Name:        "discover-variables",
-		HandlerFunc: topologyMutationHandlers.DoDiscoverVariables,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
-	}
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.GeneratePatches,
-		Name:        "generate-patches",
-		HandlerFunc: topologyMutationHandlers.DoGeneratePatches,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
-	}
-	if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
-		Hook:        runtimehooksv1.ValidateTopology,
-		Name:        "validate-topology",
-		HandlerFunc: topologyMutationHandlers.DoValidateTopology,
-	}); err != nil {
-		setupLog.Error(err, "error adding handler")
-		return err
+	allHandlers := []handlers.NamedHandler{servicelbgc.New(client)}
```

Comment:
A future improvement would be for the Handler to register itself, similar to the SchemeBuilder pattern [and now this](https://github.com/kubernetes-sigs/cluster-api/blob/2061cabe56774afa2627cb15f155ed12cf1fc9ea/api/v1beta1/machine_types.go#L309-L311)

Reply:
Great idea, and one I was going to implement but wasn't sure where the handlers would actually be registered.